src/lambda/wizards/samDeployWizard.ts

```diff
@@ -340,14 +345,35 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
     }
 
     /**
-     * Retrieves an S3 Bucket to deploy to from the user.
+     * Prompts user to select a bucket from a regionalized list of S3 buckets
+     * Caches buckets for the entire Wizard run the first time it hits this step
+     * Allows the user to go back a step if the selected region doesn't have any buckets
      *
      * @param initialValue Optional, Initial value to prompt with
      *
-     * @returns S3 Bucket name. Undefined represents cancel.
+     * @returns S3 Bucket name. Undefined represents back.
      */
     public async promptUserForS3Bucket(selectedRegion: string, initialValue?: string): Promise<string | undefined> {
-        const inputBox = input.createInputBox({
+        const s3Client = new DefaultS3Client(selectedRegion)
+        let listBucketsResponse: S3.ListBucketsOutput | undefined
+        // does the cache have any entries? If so, don't list buckets
+        if (this.regionBucketMap.size === 0) {
+            listBucketsResponse = await s3Client.listBuckets()
+        }
+        let buckets: vscode.QuickPickItem[] | undefined
+
+        const noBuckets = localize(
```

Comment:
These two consts can probably go up near the top of the file.

Reply:
`noBuckets` needs to be here (since it uses `selectedRegion`). `loadingBuckets` doesn't need to but it's only scoped to this function. Moving them to the top of the function instead.